src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -25,6 +27,78 @@ public static class Renamer
                 newName, optionSet, cancellationToken);
         }
 
+        /// <summary>
+        /// Similar to calling <see cref="Document.WithName(string)" /> with additional changes to the solution. 
+        /// Each change is added as a <see cref="RenameDocumentAction"/> in the returned <see cref="RenameDocumentActionInfo.ApplicableActions" />.
+        /// 
+        /// Each action may individually encounter errors that prevent it from behaving correctly. Those are reported in <see cref="RenameDocumentAction.Errors"/>.
+        /// 
+        /// Current supported actions that may be returned: 
+        /// * <see cref="RenameSymbolDocumentAction"/> that will rename the type to match the document name
+        /// </summary>
+        public static async Task<RenameDocumentActionInfo> RenameDocumentNameAsync(
+            Document document,
+            string newDocumentName,
+            OptionSet optionSet,
+            CancellationToken cancellationToken = default)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (newDocumentName == null)
+            {
+                throw new ArgumentNullException(nameof(newDocumentName));
+            }
+
+            var actions = new ArrayBuilder<RenameDocumentAction>();
+
+            if (!newDocumentName.Equals(document.Name))
+            {
+                var renameAction = await RenameSymbolDocumentAction.CreateAsync(document, newDocumentName, optionSet, cancellationToken).ConfigureAwait(false);
```

Comment:
See other comments: seems like this CreateAsync() should return null if there's no symbol to rename, since then the caller can see there's no actions in the list and know there's nothing to do.